src/html/classic/ng/src/web/pages/extras/trashcanpage.js

```diff
@@ -0,0 +1,493 @@
+    gmp.trashcan.delete(entity).then(this.getTrash);
+  }
+
+  handleEmptyTrashcan() {
+    const {gmp} = this.props;
+    gmp.trashcan.emptyTrashcan().then(this.getTrash);
+  }
+
+  createContentRow(type, count) {
+    return (
+      <TableRow key={type}>
+        <TableData>
+          <InnerLink to={type}>{type}</InnerLink>
+        </TableData>
+        <TableData>
+          {count}
+        </TableData>
+      </TableRow>
+    );
+  }
+
+  createContentsTable(trash) {
+    return (
+      <TableBody>
+        {this.createContentRow('Agents',
+          trash.get_agents_response.agent_count.toString())}
+        {this.createContentRow('Alerts',
+          trash.get_alerts_response.alert_count.toString())}
+        {this.createContentRow('Configs',
+          trash.get_configs_response.config_count.toString())}
+        {this.createContentRow('Credentials',
+          trash.get_credentials_response.credential_count.toString())}
+        {this.createContentRow('Filters',
+          trash.get_filters_response.filter_count.toString())}
+        {this.createContentRow('Groups',
+          trash.get_groups_response.group_count.toString())}
+        {this.createContentRow('Notes',
+          trash.get_notes_response.note_count.toString())}
+        {this.createContentRow('Overrides',
+          trash.get_overrides_response.override_count.toString())}
+        {this.createContentRow('Permissions',
+          trash.get_permissions_response.permission_count.toString())}
+        {this.createContentRow('Port Lists',
+          trash.get_port_lists_response.port_list_count.toString())}
+        {this.createContentRow('Report Formats',
+          trash.get_report_formats_response.report_format_count.toString())}
+        {this.createContentRow('Roles',
+          trash.get_roles_response.role_count.toString())}
+        {this.createContentRow('Scanners',
+          trash.get_scanners_response.scanner_count.toString())}
+        {this.createContentRow('Schedules',
+          trash.get_schedules_response.schedule_count.toString())}
+        {this.createContentRow('Tags',
+          trash.get_tags_response.tag_count.toString())}
+        {this.createContentRow('Targets',
+          trash.get_targets_response.target_count.toString())}
+        {this.createContentRow('Tasks',
+          trash.get_tasks_response.task_count.toString())}
+      </TableBody>
+    );
+  };
+
+  render() {
+    const {trash_data} = this.state;
+    const {contents_table} = this.state;
+    let agent_list;
+    let alert_list;
+    let scan_config_list;
+    let credential_list;
+    let filter_list;
+    let group_list;
+    let note_list;
+    let override_list;
+    let permission_list;
+    let port_list_list;
+    let report_format_list;
+    let role_list;
+    let scanner_list;
+    let schedule_list;
+    let tag_list;
+    let target_list;
+    let task_list;
+
+    if (!is_defined(trash_data)) {
+      return <Loading/>;
+    }
+    if (is_defined(trash_data)) {
```

Comment:
You can remove this if check. The render method is already returned if trash_data is not defined. Therefore this if condition is always true.

Reply:
Creating the CollectionLists shouldn't be done in the render method. The render method should only create and render react components. Creating the CollectionsLists better belongs to the command class.

Reply:
If-check is removed and the creation of the CollectionLists is done in the command class now.